### PR TITLE
Clean Voting code, remove direct rpc call from voting dialog

### PR DESCRIPTION
**Pull request by sedited**
This pull request  cleans some of the Smart Contract/voting code. It is part of the greater cleanup that needs to be done in order to rebase to a more up to date codebase. 
In future all the Gridcoin Smart Contract related code should go in contract/  , voting specific code was placed in polls.cpp /.h . Some of the rpc calls for the Smart Contract stuff are now in rpccontract.cpp . 
Some indenting and whitespace fixes were done as well. 
Previously the votingdialog made a direct rpc call to execute a vote. This is not really needed and the rest of the qt wallet does not use this rpc abstraction either (except for the rpc console). I suspect the current gui voting is broken anyway, so this would fix it. 
Instead of doubles, integers are now used to collect the number of days the poll will be active for and to declare the accepted Sharetypes.

### CODE REVIEW

**Comment by tomasbrod** [comment]
I do not like the introduction of another directory.

**Comment by sedited** [comment]
It's inspired by how bitcoin does it. Code with a specific use case is put into a separate subdirectory.

File: src/contract/polls.cpp [hunk 1]
```diff
@@ -0,0 +1,627 @@
...
+    boost::to_lower(email);
+    GlobalCPUMiningCPID.email = email;
+    GlobalCPUMiningCPID.lastblockhash = GlobalCPUMiningCPID.cpidhash;
+    if (!PollExists(sTitle))
+        return std::make_pair("Error", "Poll does not exist.");
+    else
+    {
+        if (PollExpired(sTitle))
+            return std::make_pair("Error", "Sorry, Poll is already expired.");
+        else
```
Comment (denravonska) [inline src/contract/polls.cpp]: I know you just moved the functions, but can you drop the else on this one to reduce one depth?

**Comment by denravonska** [comment]
Can you do a reindent of all the code in the new files? Better do it now when the diff is already large.

File: src/contract/polls.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+#include "json/json_spirit_writer_template.h"
+#include "json/json_spirit_value.h"
+#include "json/json_spirit_utils.h"
+
+#include "main.h"
+#include "bitcoinrpc.h"
+#include "init.h" // for pwalletMain
+#include "block.h"
+
+#include <boost/filesystem.hpp>
```
Comment (denravonska) [inline src/contract/polls.h]: I think you can remove most of the includes from this file.

File: src/Makefile.am [hunk 3]
```diff
@@ -115,13 +116,21 @@ GRIDCOIN_CORE_CPP = addrman.cpp \
...
 	sync.cpp \
-        tally.cpp \
+    tally.cpp \
 	txdb-leveldb.cpp \
 	util.cpp \
 	version.cpp \
 	wallet.cpp \
 	walletdb.cpp
 
+GRIDCOIN_CONTRACT_H = \
```
Comment (tomasbrod) [inline src/Makefile.am]: This separation from GRIDCOIN_CORE_*, quite useless. Only adds confusion where it should be added to build options.

File: src/contract/contract.cpp [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+std::string AddContract(std::string sType, std::string sName, std::string sContract)
+{
+    std::string sPass = (sType=="project" || sType=="projectmapping" || sType=="smart_contract") ? GetArgument("masterprojectkey", msMasterMessagePrivateKey) : msMasterMessagePrivateKey;
+    std::string result = AddMessage(true,sType,sName,sContract,sPass,AmountFromValue(1),.00001,"");
+    return result;
+}
+
+bool SignBlockWithCPID(const std::string& sCPID, const std::string& sBlockHash, std::string& sSignature, std::string& sError, bool bAdvertising=false)
```
Comment (tomasbrod) [inline src/contract/contract.cpp:100]: This function is more releated to beacon, or miner, than contracts.

File: src/miner.cpp [hunk 5]
```diff
@@ -35,8 +36,6 @@ std::string SerializeBoincBlock(MiningCPID mcpid);
 bool LessVerbose(int iMax1000);
 
 int64_t GetRSAWeightByBlock(MiningCPID boincblock);
-bool SignBlockWithCPID(const std::string& sCPID, const std::string& sBlockHash, std::string& sSignature, std::string& sError, bool bAdvertising = false);
```
Comment (tomasbrod) [inline src/miner.cpp:38]: :+1 for removing these scattered declarations

File: src/Makefile.am (hunk 3 above)
Comment (sedited) [reply src/Makefile.am]: I am planning too far ahead, I'll move it back GRIDCOIN_CORE_*

File: src/contract/contract.cpp (hunk 4 above)
Comment (sedited) [reply src/contract/contract.cpp:100]: Is it fine if I leave it there for now, and group it back, when I get to moving the beacon and miner code as well?

Comment (tomasbrod) [reply src/contract/contract.cpp:100]: iirc there is already a beacon.cpp file, but I guess it is OK.

File: src/contract/polls.cpp [hunk 6]
```diff
@@ -0,0 +1,618 @@
...
+bool WalletOutOfSync();
+
+std::pair<std::string, std::string> CreatePollContract(std::string sTitle, int days, std::string sQuestion, std::string sAnswers, int iSharetype, std::string sURL)
+{
+    if (pwalletMain->IsLocked())
+        return std::make_pair("Error", "Please fully unlock the wallet first.");
+    else if (fWalletUnlockStakingOnly)
+        return std::make_pair("Error", "Wallet is unlocked for staking only, needs to be fully unlocked for voting");
+    else if (WalletOutOfSync())
+        return std::make_pair("Error", "Wallet is not synced, you need to be synced in order to create a poll");
```
Comment (tomasbrod) [inline src/contract/polls.cpp:30]: These returnvalue error indicators should be replaced later by exceptions. Not needed fot this PR.

**Review by tomasbrod — APPROVED** [review]

**Pull request merged**